MAINTAINING.md

```diff
@@ -0,0 +1,184 @@
+branch. Check the 
+[Micronaut Module Versioning](https://github.com/micronaut-projects/micronaut-core/wiki/Micronaut-Module-Versioning)
+document for more information.
+
+Before merging pull requests, it is really important to ensure they target the correct branch, so that in the next
+patch/minor release we don't leak breaking changes. Check the
+[Micronaut Module Branch Naming](https://github.com/micronaut-projects/micronaut-core/wiki/Micronaut-Module-Branch-Naming)
+document for more information.
+
+Note that 
+[Micronaut Core and Starter](https://github.com/micronaut-projects/micronaut-core/wiki/Micronaut-Core-and-Starter-Branching-Strategy)
+follow a slightly different strategy.
+
+### Automated pull requests
+
+#### Dependency upgrades
+
+All Micronaut repos have 2 dependency upgrade checking mechanism:
+
+1. Dependabot: it has the advantage that it performs dependency upgrades not only on build dependencies, but also on
+   GitHub Actions workflows. On the other hand, its biggest downside is that it's unable to find newer versions for
+   those defined in `gradle.properties`. It will also send different PRs for the same version upgrade if the artifact ID
+   is different. For example, if you have `com.example:client:1.0` and `com.example:server:1.0`, and a new 1.1 version
+   arrives for both, it will send 2 PRs, where they should both be upgraded at the same time.
+   
+2. To overcome those disadvantages, we have our own dependency upgrade solution based on the
+   [Gradle Use Latest Versions Plugin](https://github.com/patrikerdes/gradle-use-latest-versions-plugin). It runs daily
+   during weekdays.
+   
+When an upgrade to a new version arrives, we need to be careful when merging, so that we don't introduce an
+unnecessary upgrade burden on our users. Read the
+[Module Upgrade Strategy](https://github.com/micronaut-projects/micronaut-core/wiki/Module-Upgrade-Strategy) for more
+information.
+
+Note that if a new version arrives and we are not ready yet to do the upgrade, you need to
+[pin the old version](https://github.com/micronaut-projects/micronaut-build/#configuration-options), because otherwise,
+Dependabot and our workflow will keep sending PRs. You should also create an issue to upgrade so that it's not forgotten.
+
+#### Files sync
+
+We have a [template repo](https://github.com/micronaut-projects/micronaut-project-template) that we use as the single
+source of truth for certain files. It is used as a template to create new repos, and changes to certain files in the
+temaplate repo will get propagated automatically. The files propagated are:
+
+* Workflow files (`.github/workflows/*`). They are copied using rsync.
+* Dependabot configuration (`.github/dependabot.yml`).
+* Gradle wrapper.
+* `.gitignore`.
+* `ISSUE_TEMPLATE.md`, `LICENSE`, `MAINTAINING.md`, `config/HEADER` and `config/spotless.license.java`.
+* Checkstyle's `config/checkstyle/checkstyle.xml` and `config/checkstyle/suppressions.xml`.
```

Comment:
I'd add somewhere here that there are repositories that have slightly different workflow files like `gcp` and `kubernetes`, basically because of different jdk versions. In those cases, before merging the "upgrade workflow" PR we need to manually revert some changes and then apply the new changes.